src/java.base/share/classes/java/util/ResourceBundle.java

```diff
@@ -1696,7 +1696,13 @@ private static ResourceBundle getBundleImpl(Module callerModule,
         Reference.reachabilityFence(callerModule);
         Reference.reachabilityFence(module);
 
-        return bundle;
+        //The OpenJ9 GC might collect the loader before we return here. This prevents that.
+        try {
+            return bundle;
+        } finally {
+            //Should never be true. Using the loader here prevents it being GC'd.
+            if (baseName.equals(loader)) throw new Error("Unexpected error.");
```

Comment:
This could use == instead of .equals(), which should be a faster test.

Reply:
This should use reachabilityFence in JDK9+: https://docs.oracle.com/javase/9/docs/api/java/lang/ref/Reference.html#reachabilityFence-java.lang.Object-

Aka `keepAlive`

Reply:
Incomparable object. Will try Dan's suggestion of reachabilityFence.